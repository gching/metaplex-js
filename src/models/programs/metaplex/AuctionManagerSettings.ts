import { BorshSchema } from '../../utils/types';
import { WinningConfig } from './WinningConfig';
import { ParticipationConfig } from './ParticipationConfig';

export class AuctionManagerSettings {
  static SCHEMA: BorshSchema = new Map([
    [
      AuctionManagerSettings,
      {
        kind: 'struct',
        fields: [
          ['winningConfigs', [WinningConfig]],
          [
            'participationConfig',
            { kind: 'option', type: ParticipationConfig },
          ],
        ],
      },
    ],
    ...WinningConfig.SCHEMA,
    ...ParticipationConfig.SCHEMA,
  ]);

  winningConfigs: WinningConfig[] = [];
  participationConfig?: ParticipationConfig = undefined;

  constructor(args?: AuctionManagerSettings) {
    Object.assign(this, args);
  }
}
