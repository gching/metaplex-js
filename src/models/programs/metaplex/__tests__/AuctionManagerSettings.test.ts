import { AuctionManagerSettings } from '../AuctionManagerSettings';

describe('AuctionManagerSettings', () => {
  it('should instantiate with the correct values', () => {
    const auctionManagerSettings = new AuctionManagerSettings({
      winningConfigs: [],
    });

    expect(auctionManagerSettings).toMatchSnapshot();
  });
});
