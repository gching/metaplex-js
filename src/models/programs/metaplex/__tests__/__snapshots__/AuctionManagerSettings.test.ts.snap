// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuctionManagerSettings should instantiate with the correct values 1`] = `
AuctionManagerSettings {
  "participationConfig": undefined,
  "winningConfigs": Array [],
}
`;
